### PR TITLE
Improve log handling

### PR DESCRIPTION
**Pull request by boekkooi-fresh**
This PR fixes some minor issues and changes the `log` implementation to be similar to the [spec ](https://github.com/opentracing/specification/blob/master/specification.md#log-structured-data)

### CODE REVIEW

File: src/BasicTracer/Propagation/TextMapPropagator.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace HelloFresh\BasicTracer\Propagation;
 
 use HelloFresh\BasicTracer\Exception\ExtractionException;
+use HelloFresh\BasicTracer\Exception\InjectionException;
```
Comment (nenadstojanovikj) [inline src/BasicTracer/Propagation/TextMapPropagator.php]: Is this used here?

File: src/BasicTracer/Span.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 namespace HelloFresh\BasicTracer;
 
+use HelloFresh\BasicTracer\Exception\SpanStateException;
```
Comment (nenadstojanovikj) [inline src/BasicTracer/Span.php]: Is this used?

File: src/BasicTracer/Propagation/TextMapPropagator.php (hunk 1 above)
Comment (nenadstojanovikj) [inline src/BasicTracer/Propagation/TextMapPropagator.php]: nvm

**Comment by nenadstojanovikj** [comment]
:+1:

**Comment by emilyclaudet** [comment]
+1

**Pull request merged**